src/System.Windows.Forms.Primitives/src/Interop/WinFormsComWrappers.IEnumFORMATETCVtbl.cs

```diff
@@ -0,0 +1,97 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.ComTypes;
+
+internal partial class Interop
+{
+    internal unsafe partial class WinFormsComWrappers
+    {
+        internal static class IEnumFORMATETCVtbl
+        {
+            public static IntPtr Create(IntPtr fpQueryInterface, IntPtr fpAddRef, IntPtr fpRelease)
+            {
+                IntPtr* vtblRaw = (IntPtr*)RuntimeHelpers.AllocateTypeAssociatedMemory(typeof(IEnumFORMATETCVtbl), IntPtr.Size * 7);
+                vtblRaw[0] = fpQueryInterface;
+                vtblRaw[1] = fpAddRef;
+                vtblRaw[2] = fpRelease;
+                vtblRaw[3] = (IntPtr)(delegate* unmanaged<IntPtr, int, FORMATETC*, int*, int>)&Next;
+                vtblRaw[4] = (IntPtr)(delegate* unmanaged<IntPtr, int, int>)&Skip;
+                vtblRaw[5] = (IntPtr)(delegate* unmanaged<IntPtr, int>)&Reset;
+                vtblRaw[6] = (IntPtr)(delegate* unmanaged<IntPtr, IntPtr*, int>)&Clone;
+
+                return (IntPtr)vtblRaw;
+            }
+
+            [UnmanagedCallersOnly]
+            private static int Next(IntPtr thisPtr, int celt, FORMATETC* rgelt, int* pceltFetched)
+            {
+                IEnumFORMATETC instance = ComInterfaceDispatch.GetInstance<IEnumFORMATETC>((ComInterfaceDispatch*)thisPtr);
+                FORMATETC[] elt = new FORMATETC[celt];
+                int[] celtFetched = new int[1];
+                var result = instance.Next(celt, elt, pceltFetched == null ? null! : celtFetched);
+                for (var i = 0; i < celt; i++)
+                {
+                    rgelt[i] = elt[i];
+                }
+
+                if (pceltFetched != null)
+                {
+                    *pceltFetched = celtFetched[0];
+                }
+
+                return result;
+            }
+
+            [UnmanagedCallersOnly]
+            private static int Skip(IntPtr thisPtr, int celt)
+            {
+                IEnumFORMATETC instance = ComInterfaceDispatch.GetInstance<IEnumFORMATETC>((ComInterfaceDispatch*)thisPtr);
+                try
+                {
+                    return instance.Skip(celt);
+                }
+                catch (Exception ex)
+                {
+                    return ex.HResult;
+                }
+            }
+
+            [UnmanagedCallersOnly]
+            private static int Reset(IntPtr thisPtr)
+            {
+                try
+                {
+                    IEnumFORMATETC instance = ComInterfaceDispatch.GetInstance<IEnumFORMATETC>((ComInterfaceDispatch*)thisPtr);
+                    instance.Reset();
+                }
+                catch (Exception ex)
+                {
+                    return ex.HResult;
+                }
+
+                return S_OK;
```

Comment:
This makes it easier to follow the flow:
```suggestion
                try
                {
                    IEnumFORMATETC instance = ComInterfaceDispatch.GetInstance<IEnumFORMATETC>((ComInterfaceDispatch*)thisPtr);
                    instance.Reset();
                    return S_OK;
                }
                catch (Exception ex)
                {
                    return ex.HResult;
                }
```

ditto other places